Sources/PortalSwift/Components/PortalWebView.swift

```diff
@@ -177,29 +219,24 @@ public class PortalWebView: UIViewController, WKNavigationDelegate, WKScriptMess
     self.onPageStart?()
   }
 
-  /// Called when the web view finishes loading a page. It injects the Portal script into the web view.
+  /// Called when the web view finishes loading a page.
   /// - Parameters:
   ///   - webView: The WKWebView instance that finished loading.
   ///   - navigation: The navigation information associated with the event.
   public func webView(_: WKWebView, didFinish _: WKNavigation!) {
-    guard let address = portal.address else {
-      print("[PortalWebView] No address found for user. Cannot inject provider into web page.")
-      return
-    }
-
-    let javascript = self.injectPortal(
-      address: address,
-      apiKey: self.portal.apiKey,
-      chainId: String(self.portal.chainId),
-      gatewayConfig: self.portal.gatewayConfig[self.portal.chainId]!,
-      autoApprove: self.portal.autoApprove,
-      enableMpc: true
-    )
-
-    self.evaluateJavascript(javascript, sourceURL: "portal_sign")
-
     self.onPageComplete?()
   }
+  
+  /// Called when a new tab is opened.
+  public func webView(_ webView: WKWebView, createWebViewWith configuration: WKWebViewConfiguration, for navigationAction: WKNavigationAction, windowFeatures: WKWindowFeatures) -> WKWebView? {
+          if navigationAction.targetFrame == nil {
+              webView.load(navigationAction.request)
+            
+              // open in Safari
```

Comment:
Can you add a bit more to this comment when we'd want to open it in Safari?

Reply:
Updated comment!